entity-registry/src/main/java/com/linkedin/metadata/aspect/hooks/OwnershipOwnerTypes.java

```diff
@@ -0,0 +1,120 @@
+package com.linkedin.metadata.aspect.hooks;
+
+import static com.linkedin.metadata.Constants.OWNERSHIP_ASPECT_NAME;
+
+import com.linkedin.common.*;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.metadata.aspect.ReadItem;
+import com.linkedin.metadata.aspect.RetrieverContext;
+import com.linkedin.metadata.aspect.batch.ChangeMCP;
+import com.linkedin.metadata.aspect.plugins.config.AspectPluginConfig;
+import com.linkedin.metadata.aspect.plugins.hooks.MutationHook;
+import com.linkedin.util.Pair;
+import java.util.*;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Getter
+@Setter
+@Accessors(chain = true)
+public class OwnershipOwnerTypes extends MutationHook {
+  @Nonnull private AspectPluginConfig config;
+
+  protected Stream<Pair<ChangeMCP, Boolean>> writeMutation(
+      @Nonnull Collection<ChangeMCP> changeMCPS, @Nonnull RetrieverContext retrieverContext) {
+    List<Pair<ChangeMCP, Boolean>> results = new LinkedList<>();
+    for (ChangeMCP item : changeMCPS) {
+      if (aspectFilter(item)) {
+        results.add(Pair.of(item, processOwnershipAspect(item)));
+      } else {
+        results.add(Pair.of(item, false));
+      }
+    }
+    return results.stream();
+  }
+
+  private static boolean aspectFilter(ReadItem item) {
+    return item.getAspectName().equals(OWNERSHIP_ASPECT_NAME);
+  }
+
+  public static Map<String, List<Urn>> toMap(UrnArrayMap map) {
```

Comment:
This method and the next seem like very simple utility methods. I'd bet we have a util class that does this kind of translation between the pegasus maps and java maps.

Reply:
not a blocker